landingai/predict/snowflake.py

```diff
@@ -0,0 +1,115 @@
+"""Snowflake-specific adapters and helpers"""
+
+import datetime
+from typing import Any, Dict, Optional, cast
+from urllib.parse import urljoin
+
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import serialization
+from requests import Session
+from landingai.common import APIKey
+from landingai.predict.cloud import Predictor
+from landingai.predict.utils import create_requests_session
+
+
+class SnowflakeNativeAppPredictor(Predictor):
+    """Snowflake Native App Predictor, which is basically a regular cloud predictor with a different auth mechanism."""
```

Comment:
It would be good to have some sample code on the documentation... At least like the one you provided on the description of this PR

Reply:
Let's add public docss once the native app is actually available in Snowflake. Adding it before the app is available might cause some confusion.